crypto/encode_decode/encoder_pkey.c

```diff
@@ -261,7 +261,7 @@ static int ossl_encoder_ctx_setup_for_pkey(OSSL_ENCODER_CTX *ctx,
         }
     }
 
-    if (OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
+    if (data != NULL && OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
```

Comment:
I suppose this cannot happen as OSSL_ENCODER_CTX_get_num_encoders() would return 0 here in that case. But anyway...